agent/grpc-external/services/peerstream/stream_test.go

```diff
@@ -1230,22 +1232,32 @@ func TestStreamResources_Server_DisconnectsOnHeartbeatTimeout(t *testing.T) {
 
 	client.DrainStream(t)
 
+	var lastAck time.Time
 	testutil.RunStep(t, "new stream gets tracked", func(t *testing.T) {
 		retry.Run(t, func(r *retry.R) {
 			status, ok := srv.StreamStatus(testPeerID)
 			require.True(r, ok)
 			require.True(r, status.Connected)
+			require.NotNil(r, status.LastAck)
+			lastAck = *status.LastAck
 		})
 	})
```

Comment:
A mysterious 50ms after the last ACK the disconnect time is captured by advancing the lockstep clock abstraction. If the runner was just slow enough then the disconnect time is advanced and recorded BEFORE the `stream is disconnected` subtest case is entered and it captures the wrong clock value for comparison (the NEXT one).

Now we capture the ACK value and assert that the disconnect time is just some time after it.